### ISSUE TITLE
iOS text to speech reads origin instead of highlighted text

### ISSUE THREAD

**Issue by kmh287** [ampproject/amphtml#9459]
Steps:
1: Enable text to speech by going to Settings -> accessibility -> speech
2: Open an AMP page in safari (or Chrome) on an iOS device. 
3: Highlight a block of text
4: Select "Speak" to have the text read aloud

Expected:
iOS reads the highlighted text.

Actual:
iOS reads the origin name.

NOTE: I tested this on an iPhone running 10.3.3. I was also able to reproduce on XCode's simulators.

**Comment by aghassemi** [comment]
I have a feeling this is somehow related to #5750 which has WebKit bug files under https://bugs.webkit.org/show_bug.cgi?id=163911

@kmh287 Could you try this with http://output.jsbin.com/kufici and see if it repros? http://output.jsbin.com/kufici is not an AMP page but rather a minimal repro case for https://bugs.webkit.org/show_bug.cgi?id=163911

**Comment by kmh287** [comment]
@aghassemi Cannot reproduce on that page, text to speech works as normal on that page. 

I did a little digging on this and it looks like text to speech doesn't work for text inside of an iframe. If you create a simple HTML page (no AMP at all) that has text and then an iframe that contains text, when you try to have iOS read the text inside the iframe, it will instead read the text outside. It looks like what's happening for us is iOS is reading the origin name from the viewer header. 

Looks like there's a webkit bug filed [here](https://bugs.webkit.org/show_bug.cgi?id=166794), but the bug is marked as fixed.

**Comment by aghassemi** [comment]
@kmh287 Thanks! Added a comment to the WebKit bug, looks like it has not been shipped with any iOS versions yet. (BTW, that bug was created and fixed by our own @ericfs :clap:)

Closing since tracked externally.

**Comment by aghassemi** [comment]
Got a reply on the webkit bug regarding when fix will ship:  "A future version of iOS is all we can say"

**Comment by kmh287** [comment]
@aghassemi I tested on iOS 11 and it looks like this issue may be resolved.

**Comment by aghassemi** [comment]
@kmh287 Great! Thanks for the update
